service/worker/batcher/workflow.go

```diff
@@ -20,7 +20,328 @@
 
 package batcher
 
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/.gen/go/shared"
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/workflow"
+	"golang.org/x/time/rate"
+)
+
 const (
 	batcherContextKey   = "batcherContext"
 	batcherTaskListName = "cadence-sys-batcher-tasklist"
+	batchWFTypeName     = "cadence-sys-batch-workflow"
+	batchActivityName   = "cadence-sys-batch-activity"
+
+	infiniteDuration = 20 * 365 * 24 * time.Hour
+	pageSize         = 1000
+
+	// below are default values for BatchParams
+
+	defaultRPS                      = 50
+	defaultConcurrency              = 5
+	defaultAttemptsOnRetryableError = 50
+	defaultActivityHeartBeatTimeout = time.Minute
 )
+
+const (
+	BatchTypeTerminate = "terminate"
+	BatchTypeReset     = "reset"
+)
+
+type (
+	BatchParams struct {
```

Comment:
Please check and fix lint warning before merge

Reply:
done